### PR TITLE
retries, error handling, session resynchronization

### PR DESCRIPTION
**Pull request by avereha**
Issued solved in this PR:
 - improved error handling. For writes, we will get two class of error: SendError and Send and not confirmed.
 - added EAP-AKA session resynchronization
 - merged with `bart/implement-pump-plugin`
 - added retries for lost DATA packets(NACKs)

### CODE REVIEW

**Review by AdrianLxM — COMMENTED** [review]
some small kotlin remarks

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/OmnipodDashPumpPlugin.kt [hunk 1]
```diff
@@ -93,40 +103,136 @@ class OmnipodDashPumpPlugin @Inject constructor(
...
     }
 
     override val baseBasalRate: Double
-        get() = 0.0 // TODO
+        get() {
+            if (podStateManager.basalProgram == null) {
+                return 0.0
+            }
+            return podStateManager.basalProgram!!.rateAt(Date())
+        }
```
[the remark this answers is not in the document]
Comment (avereha) [reply omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/OmnipodDashPumpPlugin.kt:165]: Done.

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/OmnipodDashPumpPlugin.kt [hunk 2]
```diff
@@ -214,11 +369,189 @@ class OmnipodDashPumpPlugin @Inject constructor(
...
+        if (customCommand is CommandHandleTimeChange) {
+            return handleTimeChange()
+        }
+        if (customCommand is CommandUpdateAlertConfiguration) {
+            return updateAlertConfiguration()
+        }
+        if (customCommand is CommandPlayTestBeep) {
+            return playTestBeep()
+        }
```
[the remark this answers is not in the document]
Comment (avereha) [reply omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/OmnipodDashPumpPlugin.kt:392]: Done, rewrote as `return when(..`

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/OmnipodDashPumpPlugin.kt [hunk 3]
```diff
@@ -93,40 +103,136 @@ class OmnipodDashPumpPlugin @Inject constructor(
...
     }
 
     override fun isThisProfileSet(profile: Profile): Boolean {
-        // TODO
-        return true
+        if (podStateManager.basalProgram == null) {
+            // We don't have an active Pod. Return true here because the next Pod will use the currently active profile
+            return true
+        }
+        return podStateManager.basalProgram!! == mapProfileToBasalProgram(profile)
```
[the remark this answers is not in the document]
Comment (avereha) [reply omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/OmnipodDashPumpPlugin.kt:152]: Done.

**Pull request merged**